src/request/derive-polygon-address/DerivePolygonAddress.js

```diff
@@ -0,0 +1,90 @@
+/* global PasswordBox */
+/* global KeyStore */
+/* global Errors */
+/* global Utf8Tools */
+/* global TopLevelApi */
+/* global PolygonKey */
+
+/**
+ * @callback DerivePolygonAddress.resolve
+ * @param {KeyguardRequest.DerivePolygonAddressResult} result
+ */
+
+class DerivePolygonAddress {
+    /**
+     * @param {Parsed<KeyguardRequest.DerivePolygonAddressRequest>} request
+     * @param {DerivePolygonAddress.resolve} resolve
+     * @param {reject} reject
+     */
+    constructor(request, resolve, reject) {
+        this._request = request;
+        this._resolve = resolve;
+        this._reject = reject;
+
+        /** @type {HTMLFormElement} */
+        const $passwordBox = (document.querySelector('.password-box'));
+
+        // Create components
+
+        this._passwordBox = new PasswordBox($passwordBox, {
+            hideInput: !request.keyInfo.encrypted,
+            buttonI18nTag: 'passwordbox-continue',
+        });
+
+        // Wire up logic
+
+        this._passwordBox.on(PasswordBox.Events.SUBMIT, /** @param {string|undefined} password */ password => {
+            this._onPasswordEntered(password);
+        });
+    } // constructor
+
+    /**
+     * @param {string} [password]
+     * @returns {Promise<void>}
+     */
+    async _onPasswordEntered(password) {
+        TopLevelApi.setLoading(true);
+        const passwordBuffer = password && password.length > 0
+            ? Utf8Tools.stringToUtf8ByteArray(password)
+            : undefined;
```

Comment:
```suggestion
        const passwordBuffer = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
```
Password length check already covered by just checking for `password`.